core/runtime.rs

```diff
@@ -495,8 +487,14 @@ impl JsRuntime {
     // TODO(piscisaureus): The rusty_v8 type system should enforce this.
     state.borrow_mut().global_context.take();
 
-    // Drop v8::Global handles before snapshotting
-    std::mem::take(&mut state.borrow_mut().module_map);
+    // Overwrite existing ModuleMap to drop v8::Global handles
+    self
+      .v8_isolate()
+      .set_slot(Rc::new(RefCell::new(ModuleMap::new(
```

Comment:
Is this necessary? 
IIRC `set_slot` is generic and `v8_isolate().set_slot(())` should work.

Reply:
Wouldn't that set `()` in a new slot instead of overwriting existing slot with `Rc<RefCell<ModuleMap>>`?

Reply:
You're right. Oops.